app/stac_api/models.py

```diff
@@ -270,6 +274,51 @@ class Item(models.Model):
         null=False, blank=False, default=BBOX_CH, srid=4326, validators=[validate_geometry]
     )
 
+    class Meta:
+        unique_together = (('collection', 'name'),)
+        indexes = [
+            models.Index(fields=['name'], name='item_name_idx'),
+            # the following 3 indices are used e.g. in collection_temporal_extent
+            models.Index(fields=['properties_datetime'], name='item_datetime_idx'),
+            models.Index(fields=['properties_start_datetime'], name='item_start_datetime_idx'),
+            models.Index(fields=['properties_end_datetime'], name='item_end_datetime_idx'),
+            # created, updated, and title are "queriable" in the search endpoint
+            # see: views.py:322 and 323
+            BrinIndex(fields=['created'], name='item_created_idx'),
+            models.Index(fields=['updated'], name='item_updated_idx'),
+            models.Index(fields=['properties_title'], name='item_title_idx'),
+            # geometry is used in "filter_by_intersects", views.py:415
+            models.Index(fields=['geometry'], name='item_geometry_idx'),
+            # combination collection name and item name is used in get_item_etag
+            # views.py:58
+            # TODO: still need to figure out, how to create a multi-field index # pylint: disable=fixme
+            # with fields from two different classes..
+            # self.collection.name won't work
+            # models.Index(
+            #     fields=['self.collection.name', 'name'], name='item_coll_name_item_name_idx'
```

Comment:
Any idea anyone, how the `self.collection.name` field would be correctly referenced here in a multi-field index with fields from two different classes?

Reply:
`collection__name` worked neither

Reply:
I'm not even sure this is possible conceptually, from how relational databases work. I could imagine that you need two queries anyhow for this usecase, the first one to get the collection_id for a specific collection name (for this query you would use an index on collection name) and a second query to get the item name, searching only in items with a specific collection id (for this you would use a combined index collection_id, item_name, (which in django would be `fields=['collection','name']).

Reply:
I had similar issue with MongoDB. To solve it, we could duplicate the collection name in the item table, we will have a drawback in the write but it will improve performance in the query.

Reply:
I'll drop this specific combined index for now. Currently the potential use case for this index uses a combination of the indices `item_name_idx` and `stac_api_collection_pkey` instead. That might do as well and prevents this probably too expensive and not really needed index in this case.